contracts/peripheral/grants/VeBalGrant.sol

```diff
@@ -0,0 +1,291 @@
+    function increaseTime(uint256 to) external onlyAuth whileActive {
+        votingEscrow.increase_unlock_time(to);
+    }
+
+    // @notice Exit BPT for BAL ETH
+    function redeem() external onlyAuth whileInactive {
+        _exitBalEthPool();
+    }
+
+    /// @notice Release veBAL lock
+    function release() external onlyAuth whileInactive {
+        votingEscrow.withdraw();
+    }
+
+    /// @notice Claim BAL from the veBAL gauge
+    function claimBalAndLock() external onlyAuth whileActive {
+        IFeeDistributor(feeDistributor).claimToken(address(this), BAL);
+        _joinBalEthPool();
+        uint256 balance = BAL_ETH_BPT.balanceOf(address(this));
+        console.log(balance);
+        _increaseLock(balance);
+    }
+
+    function setApprovals() external onlyAuth {
+        WETH.approve(address(BALANCER_VAULT), type(uint256).max);
+        BAL.approve(address(BALANCER_VAULT), type(uint256).max);
+    }
+
+    function approveParties() external onlyAuth whileInactive {
+        WETH.approve(project, type(uint256).max);
+        BAL.approve(balancer, type(uint256).max);
+    }
+
+    /// @notice Forward HH voting incentives
+    function forwardIncentives(address _to) external {
+        if (active) {
+            require(msg.sender == project);
+        } else {
+            require(msg.sender == balancer);
+        }
+
+        IHiddenHand(hiddenHand).setRewardForwarding(_to);
+    }
+
+    /* ----------------------------------------------------------------
+       Project Functions
+    ---------------------------------------------------------------- */
+
+    /// @notice Vote for a gauge weight
+    function voteGaugeWeight(address gauge, uint256 weight) external {
+        if (active) {
+            require(msg.sender == project);
+        } else {
+            require(msg.sender == balancer);
+        }
+        gaugeController.vote_for_gauge_weights(gauge, weight);
+    }
+
+    /// @notice Claim fees
+    function claimFees(
+        address distro,
+        address token,
+        address to
+    ) external onlyProject whileActive {
+        require(token != address(BAL) && token != address(WETH), "!token");
+        IFeeDistributor(distro).claimToken(address(this), IERC20(token));
+        IERC20(token).transfer(to, IERC20(token).balanceOf(address(this)));
+    }
+
+    function fundWeth(uint256 _amount) external onlyProject whileActive {
+        require(_amount >= minimumProjectFunding, "!enough");
+        WETH.safeTransferFrom(project, address(this), _amount);
+    }
+
+    /* ----------------------------------------------------------------
+       Balancer Functions 
+    ---------------------------------------------------------------- */
+
+    /// @notice Queue a release to stop any new locks
+    function deactivate() external onlyBalancer whileInactive {
+        active = false;
+    }
+
+    function fundGrant(uint256 _amount, uint256 _minimumProjectFunding) external onlyBalancer whileInactive {
```

Comment:
Don't actually need this function either. There will be multiple tranches of funding so setting active and `minimumProjectFunding` here doesnt make sense.

The minimum funding should just be a social agreement. If they break it, then Balancer de-activates this and the grant is cancelled anyway. Can just remove this fn and the param